azure-pipelines/ui-automation/templates/build-msal-automation-app.yml

```diff
@@ -0,0 +1,62 @@
+parameters:
+  - name: brokerFlavor
+    type: string
+    default: 'BrokerHost'
+    values:
+      - BrokerHost
+      - BrokerMicrosoftAuthenticator
+      - BrokerCompanyPortal
+      - AutoBroker
+  - name: msalFlavor
+    type: string
+    default: 'Local'
+    values:
+      - Local
+      - Dist
+  - name: brokerSource
+    displayName: Broker App Installation Source
+    type: string
+    default: LocalApk
+    values:
+      - LocalApk
+      - PlayStore
+
+jobs:
+  - job: 'msalautomationapp'
+    displayName: Build and Publish MSAL Automation app
+    pool:
+      vmImage: ubuntu-latest
+    steps:
+      - checkout: self
+        clean: true
+        submodules: recursive
+        persistCredentials: True
+      - bash: |
+          echo "##vso[task.setvariable variable=ENV_VSTS_MVN_ANDROID_MSAL_ACCESSTOKEN]$(mvnAccessToken)"
+        displayName: Set MVN Access Token in Environment
+      - task: AzureKeyVault@2
+        displayName: 'Get Key vault AndroidAutomationRunnerAppSecret'
+        inputs:
+          azureSubscription: 'MSIDLABS_ANDROID_KV'
+          KeyVaultName: 'ADALTestInfo'
+          SecretsFilter: 'AndroidAutomationRunnerAppSecret'
+          RunAsPreJob: false
```

Comment:
my recommendation will be to start removing all local mvnAccessToken and use $(AndroidAuthClient-Internal-Maven-Access-Token) from group variables AndroidAuthClientVariables, anyway this work can be realized in another PBI for all the pipelines.

Reply:
Yes, can be explored as a separate effort for all pipelines.